test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -896,6 +961,80 @@ UCS_TEST_P(test_ucp_sockaddr, err_handle_without_err_cb)
     EXPECT_EQ(1u, sender().get_err_num());
 }
 
+UCS_TEST_SKIP_COND_P(test_ucp_sockaddr, force_close_during_rndv,
+                     (send_recv_type() != SEND_RECV_TAG), "RNDV_THRESH=0")
+{
+    constexpr size_t length = 4 * UCS_KBYTE;
+
+    listen_and_communicate(false, SEND_DIRECTION_BIDI);
+
+    mem_buffer recv_buffer(length, UCS_MEMORY_TYPE_HOST);
+    recv_buffer.pattern_fill(2, length);
+    void *rreq = recv(receiver(), recv_buffer.ptr(), length,
+                      rtag_complete_check_data_cbx,
+                      reinterpret_cast<void*>(&recv_buffer));
+
+    mem_buffer send_buffer(length, UCS_MEMORY_TYPE_HOST);
+    send_buffer.pattern_fill(1, length);
+    void *sreq = send(sender(), send_buffer.ptr(), length,
+                      send_recv_type(), scomplete_reset_data_cbx,
+                      reinterpret_cast<void*>(&send_buffer));
+
+    ucp_ep_h ep = sender().revoke_ep();
+
+    // Wait for the TAG RNDV/RTS packet sent and the request scheduled to be
+    // tracked until RNDV/ATS packet is not received from a peer
+    ucs_time_t deadline = ucs::get_deadline();
+    do {
+        ucp_worker_progress(sender().worker());
+    } while (ucs_hlist_is_empty(&ucp_ep_ext_gen(ep)->proto_reqs) &&
+             (ucs_get_time() < deadline));
+    EXPECT_FALSE(ucs_hlist_is_empty(&ucp_ep_ext_gen(ep)->proto_reqs));
+
+    std::map<uct_iface_h, uct_iface_ops_t> sender_uct_eps;
+
+    // Prevent destroying UCT endpoints from discarding to not detect error by
+    // the receiver earlier than data could invalidate by the sender
+    for (auto lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+        if (lane == ucp_ep_get_cm_lane(ep)) {
+            continue;
+        }
+
+        uct_iface_h uct_iface = ep->uct_eps[lane]->iface;
+        if (sender_uct_eps.find(uct_iface) == sender_uct_eps.end()) {
+            sender_uct_eps.emplace(uct_iface, uct_iface->ops);
```

Comment:
I would suggest to emplace then check return value

Reply:
done